stackslib/src/config/mod.rs

```diff
@@ -2134,90 +2163,106 @@ pub struct NodeConfig {
     /// Normal operation requires this to be `true`; setting to `false` will likely break consensus adherence.
-    /// This parameter cannot be set via the configuration file; it must be modified programmatically.
-    ///
-    /// Default: `true`
+    /// ---
+    /// @default: `true`
+    /// @notes:
+    ///   - This parameter cannot be set via the configuration file; it must be modified programmatically.
     pub assume_present_anchor_blocks: bool,
     /// Fault injection setting for testing purposes. If set to `Some(p)`, where `p` is between 0 and 100,
     /// the node will have a `p` percent chance of intentionally *not* pushing a newly processed block
     /// to its peers.
-    ///
-    /// Default: `None` (no fault injection).
+    /// ---
+    /// @default: `None` (no fault injection)
+    /// @notes:
+    ///   - Values: 0-100 (percentage).
     pub fault_injection_block_push_fail_probability: Option<u8>,
     /// Fault injection setting for testing purposes. If `true`, the node's chainstate database
     /// access layer may intentionally fail to retrieve block data, even if it exists,
     /// simulating block hiding or data unavailability.
-    /// This parameter cannot be set via the configuration file; it must be modified programmatically.
-    ///
-    /// Default: `false`
+    /// ---
+    /// @default: `false`
+    /// @notes:
+    ///   - This parameter cannot be set via the configuration file; it must be modified programmatically.
     pub fault_injection_hide_blocks: bool,
     /// The polling interval, in seconds, for the background thread that monitors chain liveness.
     /// This thread periodically wakes up the main coordinator to check for chain progress or
     /// other conditions requiring action.
-    ///
-    /// Default: `300` seconds (5 minutes)
+    /// ---
+    /// @default: `300` (seconds, 5 minutes)
+    /// @notes:
+    ///   - Units: seconds.
     pub chain_liveness_poll_time_secs: u64,
     /// A list of specific StackerDB contracts (identified by their qualified contract identifiers,
     /// e.g., "SP000000000000000000002Q6VF78.pox-3") that this node should actively replicate.
-    ///
-    /// Default: The initial list depends on the [`NodeConfig::miner`] and [`NodeConfig::stacker`] settings:
-    /// - If `miner = true` or `stacker = true`: Relevant system contracts (like `.miners`
-    ///   and `.signers-*`) are automatically added during configuration loading, in addition
-    ///   to any contracts specified in the configuration file.
-    /// - If `miner = false` and `stacker = false`: Empty vector `[]`.
+    /// ---
+    /// @default:
+    ///   - If [`NodeConfig::miner`] is `true` or [`NodeConfig::stacker`] is `true`, relevant system contracts
+    ///     (like `.miners`, `.signers-*`) are automatically added in addition to any contracts
+    ///     specified in the configuration file.
+    ///   - Otherwise, defaults to an empty list `[]` if not specified in the TOML.
```

Comment:
This is an interesting case here - i'm not sure a simple `@defaults` with a list is the best way to present this data (but as currently defined, it's the only option available). 

what if we defined something like a `case` statement? 
ex:
```
@defaults:
     [`NodeConfig::miner`] is `true` or [`NodeConfig::stacker`] is `true`: relevant system contracts (like `.miners`, `.signers-*`) are automatically added in addition to any contracts specified in the configuration file.
     else: defaults to an empty list `[]` if not specified in the TOML
```

a little messy just copypasting, but it would make more complex examples easier to parse/render (with better syntax rules than i'm using in my example).